Proposal-EDGI_ML_project.md

```diff
@@ -0,0 +1,162 @@
+2. Direct analysts to important changes on active websites which will help them to efficiently document and analyze relevant information.
+3. Identify missing data and information files and attempt to find archived versions on other websites.
+
+**Describe how your project meets this need** :
+
+The changes worth reviewing will tend to have some characteristic features which can be classified with the help of various classification algorithms. A considerable amount of time will be devoted to the preparation of the training dataset which is an essential task while working on Machine Learning problems.
+
+A variety of Machine Learning models will be trained on the thoroughly labeled dataset. They will also be tested to evaluate the performance by using some part of the data for validation.
+
+The project will follow the following order of tasks to fill the aforementioned needs:
+
+- Step zero will be to convert data from different sources (PageFreezer, Versionista) into a common form and storing them in one place. If it is already handled, step zero will be to understand the database schema and the data stored in it.
+- The first step will be improving the method of computing differences. This will consist of two sub-tasks:
+  - The changes can be categorized based on the type of object that has changed(text, image, hyperlinks etc.). The first sub-task will be to separate the changes into categories.
+  - The second sub-task will be to extract meaningful features from data of each type. This will include extracting information like type and length of the changed matter in a given text. If required, complex techniques can be used to extract features (NLP, Image Processing).
+- The second task will be writing scripts for different Machine Learning Classification Algorithms which will learn to prioritize the changes. Each script will consist of four basic parts:
+  - Data loading and pre-processing (if required) to convert data to input format of algorithm. Also, division of data into training and validation sets.
+  - Machine Learning model creation using libraries like numpy, scikit-learn, [Tensorflow](https://www.tensorflow.org/) etc. .
+  - Evaluation of model performance (using evaluation metrics) and creation of visualizations from the results of the validation set.
+  - Storing the trained model by serializing the model with the help of modules like pickle, joblib.
+- The final task will include the following:
+  - Retrieving the trained models by deserializing the stored files and using them to prioritize new changes.
+  - Providing a list of prioritized changes to analysts for reviewing and adding annotations. An important step here will be to add correctly predicted cases to the training dataset. The model will be re-trained only after a sufficient amount of correctly predicted data is added.
+
+**Milestones/Timeline** :
+

```

Comment:
@suchthis Took your advice and modified my timeline and goals for the summer a bit. The commit shows that the whole file has changed but I've only edited a few lines in the timeline. This is because of the way I've uploaded an updated version. I'm new to Github, please forgive me for this.